mountpoint-s3/src/prefetch/caching_stream.rs

```diff
@@ -247,23 +314,24 @@ where
                     self.block_offset,
                     &self.cache_key,
                 );
-                self.part_queue_producer.push(Ok(make_part(
-                    self.buffer.clone(),
-                    self.block_index,
-                    self.block_offset,
-                    block_size,
-                    &self.cache_key,
-                    &self.original_range,
-                )));
                 self.block_index += 1;
                 self.block_offset += block_size;
                 self.buffer = ChecksummedBytes::default();
             }
         }
+        Ok(())
+    }
 
+    /// Flush remaining data in the buffer to the cache. This can be called to write the last
+    /// block for the object.
+    fn flush(self) {
```

Comment:
`flush` should not be needed anymore as a separate method. We should be able to move this logic at the end of `compose_parts`, right?

Reply:
Yes, we can move it into `compose_parts` now.